### PR TITLE
DAT-3711 Adding sections to the infobox builder

### PR DESCRIPTION
**Pull request by SebastianMarzjan**
## Links
* http://wikia-inc.atlassian.net/browse/DAT-3711

## Description
Add the ability to add section header elements within the infobox builder

## Reviewers
@bognix @tomasznapieralski @dianafa

### CODE REVIEW

File: front/main/app/components/infobox-builder-item-section-header.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import Ember from 'ember';
+import InfoboxBuilderItemMixin from '../mixins/infobox-builder-item';
+
+export default Ember.Component.extend(
+	InfoboxBuilderItemMixin, {
+		init() {
+			this._super(...arguments);
+			this.tagName = 'h3';
```
Comment (bognix) [inline front/main/app/components/infobox-builder-item-section-header.js]: `tagName` and `classNames` can (and should) be defined outside `init`

Those properties are ember standard properties which are handled by ember so there is no need to worry about sharing the state between instances of this component.

File: front/main/app/components/infobox-builder-item-section-header.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+			this.classNames = ['pi-item', 'pi-header', 'pi-secondary-font', 'pi-item-spacing',
+				'pi-secondary-background'];
+
+		},
+		header: Ember.computed.oneWay('item.data')
```
Comment (bognix) [inline front/main/app/components/infobox-builder-item-section-header.js]: minor fix - you can consider using http://emberjs.com/api/classes/Ember.computed.html#method_readOnly

**Comment by SebastianMarzjan** [comment]
@Warkot @vforge - CR changes added, feel encouraged to review again :)

File: front/main/app/components/infobox-builder-item-section-header.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import Ember from 'ember';
+import InfoboxBuilderItemMixin from '../mixins/infobox-builder-item';
+
+export default Ember.Component.extend(
+
```
Comment (vforge) [inline front/main/app/components/infobox-builder-item-section-header.js]: not needed `\n`

File: front/main/app/models/infobox-builder.js [hunk 4]
```diff
@@ -391,6 +412,21 @@ InfoboxBuilderModel.reopenClass({
...
+	 * @desc Overrides some properties of given header object with additional
+	 * data, obtained from already existing template
+	 *
+	 * @param {Object} item item to extend
+	 * @param {Object} itemData additional data
+	 * @returns {Object}
+	 */
+	extendHeaderData(item, itemData) {
+		if (itemData) {
+			// TODO: add support for collapsible attribute - DAT-3732
```
Comment (vforge) [inline front/main/app/models/infobox-builder.js]: Please use `@todo` instead of `TODO:`

**Comment by vforge** [comment]
Very minor styling comments, LGTM otherwise.

**Comment by SebastianMarzjan** [comment]
Thanks!

**Comment by vforge** [comment]
Awesome, LGTM!

File: front/main/app/models/infobox-builder.js [hunk 5]
```diff
@@ -391,6 +412,22 @@ InfoboxBuilderModel.reopenClass({
 		}
 
 		return item;
+	},
+
+	/**
+	 * @desc Overrides some properties of given header object with additional
```
Comment (Warkot) [inline front/main/app/models/infobox-builder.js]: It's just a minor thing, but according to our guidelines, `@desc` is omitted → https://github.com/Wikia/guidelines/blob/master/JavaScript/CodingConventions.md#comments.

File: front/main/app/models/infobox-builder.js [hunk 6]
```diff
@@ -391,6 +412,22 @@ InfoboxBuilderModel.reopenClass({
...
+	 * data, obtained from already existing template
+	 *
+	 * @param {Object} item item to extend
+	 * @param {Object} itemData additional data
+	 * @returns {Object}
+	 */
+	extendHeaderData(item, itemData) {
```
Comment (Warkot) [inline front/main/app/models/infobox-builder.js]: Try reading this function aloud—sounds like poetry! :smile:

**Comment by Warkot** [comment]
One negligible comment about JSDoc and LGTM. Just be aware that you have to merge `dev` in. And :+1: for tests! :smiley_cat:

File: front/main/app/models/infobox-builder.js (hunk 6 above)
Comment (dianafa) [inline front/main/app/models/infobox-builder.js]: :D but kinda bad one... :P

**Pull request merged**